askama/src/filters/mod.rs

```diff
@@ -511,6 +511,39 @@ pub fn wordcount(s: impl ToString) -> Result<usize, Infallible> {
     wordcount(s.to_string())
 }
 
+/// Returns a titlecased version of the value. I.e. words will start with uppercase letters, all
+/// remaining characters are lowercase.
+#[inline]
+pub fn title(s: impl ToString) -> Result<String, Infallible> {
+    let s = s.to_string();
+    let mut need_capitalization = true;
+
+    // Sadly enough, we can't mutate a string when iterating over its chars, likely because it could
+    // change the size of a char, "breaking" the char indices.
+    Ok(s.chars().fold(String::new(), |mut acc, c| {
+        if c.is_whitespace() {
+            acc.push(c);
+            need_capitalization = true;
+        } else if need_capitalization {
+            if !c.is_uppercase() {
```

Comment:
I think the best formulation of this would use `match` instead of `if` and `push_str()` or `extend()` instead of the loop.

Reply:
:+1: